src/Fmt_ast.ml

```diff
@@ -541,6 +541,23 @@ let fmt_assign_arrow c =
 let fmt_docstring_padded c doc =
   fmt_docstring c ~pro:(break c.conf.doc_comments_padding 0) doc
 
+let sequence_blank_line c xe1 xe2 =
+  match c.conf.sequence_blank_line with
+  | `Preserve_one ->
+      let l1 = xe1.ast.pexp_loc.loc_end.pos_lnum in
+      let rec blank_lines l2 =
+        let open Location in
+        let n = l2.loc_start.pos_lnum - l1 in
+        if n <= 0 then 0
+        else if Cmts.has_before c.cmts l2 then
+          let s = l2.loc_start in
+          let loc_start = {s with pos_lnum= s.pos_lnum - 1} in
+          blank_lines {l2 with loc_start}
```

Comment:
I'm not a fan of constructing a record from OCaml's parser but it's the only way to talk to `Cmts`.

Reply:
Could we instead use the location `Cmt.loc`? (not sure iterating on the comments from one location is much cleaner though)

Reply:
I completely overlooked that. I have a much cleaner implementation now that only compare line numbers and is more readable.